File: backend/app/prompts/review.py
"""Review Agent Prompt 模板"""

SYSTEM_PROMPT = """你是一位资深代码审查员 (Review Agent)。你的职责是对代码进行 Code Review，发现问题并给出改进建议。

## 输出格式要求

你必须输出严格的 JSON 格式，不要包含任何其他文字说明。JSON Schema 如下：

{
  "comments": [
    {
      "file_path": "文件路径",
      "line_number": 10,
      "severity": "info|warning|error",
      "category": "security|performance|style|bug|suggestion",
      "message": "问题描述",
      "suggestion": "修复建议（可选代码片段）"
    }
  ],
  "summary": "整体评价",
  "score": 85,
  "passed": true
}

## Review 关注点

1. 安全性：SQL注入、XSS、敏感信息泄露
2. 性能：N+1查询、不必要的循环、缺少索引
3. 代码质量：命名规范、重复代码、过长函数
4. 错误处理：异常捕获、边界情况
5. 类型安全：类型注解、空值处理
"""

USER_PROMPT_TEMPLATE = """请对以下代码进行 Code Review：

## 任务
{title}

## 代码内容
{description}

## 上下文
{context}
"""

File: frontend/src/api/review.ts
import request from '@/utils/request'

export interface ReviewComment {
  id: number
  task_id: number
  node_id: string | null
  file_path: string | null
  line_number: number | null
  severity: 'info' | 'warning' | 'error'
  category: string | null
  message: string
  suggestion: string | null
}

export interface ReviewSummary {
  total: number
  errors: number
  warnings: number
  infos: number
  comments: ReviewComment[]
}

export interface TestResultItem {
  id: number
  task_id: number
  node_id: string | null
  test_name: string
  status: 'passed' | 'failed' | 'error' | 'skipped'
  duration_ms: number
  error_msg: string | null
}

export interface TestSummary {
  total: number
  passed: number
  failed: number
  errors: number
  skipped: number
  results: TestResultItem[]
}

export function getTaskReview(taskId: number) {
  return request.get<any, ReviewSummary>(`/tasks/${taskId}/review`)
}

export function getTaskTests(taskId: number) {
  return request.get<any, TestSummary>(`/tasks/${taskId}/tests`)
}
